.dotnet/tests/Embeddings/EmbeddingsMockTests.cs

```diff
@@ -0,0 +1,219 @@
+﻿using System;
+using System.ClientModel;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using NUnit.Framework;
+using OpenAI.Embeddings;
+using OpenAI.Tests.Utility;
+
+namespace OpenAI.Tests.Embeddings;
+
+[TestFixture(true)]
+[TestFixture(false)]
+[Parallelizable(ParallelScope.All)]
+[Category("Smoke")]
+public partial class EmbeddingsMockTests : SyncAsyncTestBase
+{
+    private static readonly ApiKeyCredential s_fakeCredential = new ApiKeyCredential("key");
+
+    public EmbeddingsMockTests(bool isAsync)
+        : base(isAsync)
+    {
+    }
+
+    [Test]
+    public async Task GenerateEmbeddingDeserializesVector()
```

Comment:
I'm adding mock tests for the properties we cannot validate deterministically. For example, while we can validate some things about the returned `Vector` property, we can't ensure deserialization works as intended since there's no expected value to compare to. We're adding mock tests for these properties to ensure deserialization works.